### ISSUE TITLE
windows_unc_path(path) is setting this \\?\  -  /sbin/mount.vboxsf: mounting failed with the error: No such file or directory

### ISSUE THREAD

**Issue by pearcec** [mitchellh/vagrant#8357]
First GuestAdditions are installed.  I am running from Powershell.  For some reason we add \\?\ to the path.  When I alter the code to simply return the path it works.  I am not sure why this is added or what purpose it should serve, but it causes a failure for me.

https://github.com/mitchellh/vagrant/blob/master/lib/vagrant/util/platform.rb#L206 

### Vagrant version
Vagrant 1.9.2

### Host operating system
Windows 7

### Guest operating system
RHEL 7.x

### Vagrantfile
```ruby
Vagrant.configure("2") do |config|

  lin_box    = '[redacted]'
  win_box    = '[redacted]'
  linbox_url = '[redacted]'
  winbox_url = '[redacted]'

  # Skip the update check, it just wastes time
  config.vm.box_check_update = false

  # Use hostmanager
  config.hostmanager.enabled = true
  config.hostmanager.manage_guest = true
  config.hostmanager.ignore_private_ip = false

  # Puppet master
  config.vm.define "puppet", primary: true do |puppet|
    puppet.vm.box     = lin_box
    puppet.vm.box_url = linbox_url

    # Networking
    puppet.vm.hostname = "puppet"
    puppet.vm.network :private_network, :ip => '10.10.10.10'

    # Puppet Master
    puppet.vm.provider :virtualbox do |vbox|
      vbox.customize [
        'modifyvm', :id,
        '--name', 'puppet',
      ]
      vbox.memory = 2048
      vbox.cpus = 2
    end

    puppet.vm.provision "shell", path: "scripts/provision.sh"

    # Provision puppet, with puppet
    puppet.vm.provision "puppet" do |puppet|
      puppet.environment_path = "environments"
      puppet.environment = "sandbox"
    end

    puppet.vm.provision "shell", path: "scripts/r10k.sh"

    # Host file configuration
    puppet.hostmanager.aliases = %w(puppet)
  end

  # Agent1 configuration (Linux)
  config.vm.define 'agent1' do |agent1|
    agent1.vm.box     = lin_box
    agent1.vm.box_url = linbox_url

    # Networking
    agent1.vm.hostname = "agent1"
    agent1.vm.network :private_network, :ip => '10.10.10.20'

    # VM Configuration
    agent1.vm.provider :virtualbox do |vbox|
      vbox.customize [
        'modifyvm', :id,
        '--name', 'agent1',
      ]
      vbox.memory = 512
      vbox.cpus = 1
    end

    agent1.vm.provision "shell", path: "scripts/provision.sh"

    # Host file configuration
    agent1.hostmanager.aliases = %w(agent1)
  end

  # Agent2 configuration (Windows)
  config.vm.define 'agent2' do |agent2|
    agent2.vm.box     = win_box
    agent2.vm.box_url = winbox_url

    # Networking
    agent2.vm.hostname = "agent2"
    agent2.vm.network :private_network, :ip => '10.10.10.30'

    # VM Configuration
    agent2.vm.provider :virtualbox do |vbox|
      vbox.customize [
        'modifyvm', :id,
        '--name', 'agent2',
      ]
      vbox.memory = 512
      vbox.cpus = 1
    end

    agent2.vm.provision "shell", path: "scripts/provision.ps1"

    # Host file configuration
    agent2.hostmanager.aliases = %w(agent2)
  end
end
```

### Debug output
```
==> puppet: Mounting shared folders...
    puppet: /vagrant => C:/Users/hccpp302/Documents/projects/vagrant-puppet4-oss
Vagrant was unable to mount VirtualBox shared folders. This is usually
because the filesystem "vboxsf" is not available. This filesystem is
made available via the VirtualBox Guest Additions and kernel module.
Please verify that these guest additions are properly installed in the
guest. This is not a bug in Vagrant and is usually caused by a faulty
Vagrant box. For context, the command attempted was:

mount -t vboxsf -o uid=900,gid=1000 vagrant /vagrant

The error output from the command was:

/sbin/mount.vboxsf: mounting failed with the error: No such file or directory
```

### Expected behavior
It works

### Actual behavior
It doesn't mount the /vagrant directory

### Steps to reproduce
1. vagrant up

**Comment by chrisroberts** [comment]
Dup of #8352. Thanks for the report!

**Comment by cbj4074** [comment]
VirtualBox 5.1.16 breaks "Shared Folder" and NFS mounts (at least) on Windows 10 hosts (at least)



### Vagrant version
1.9.2

### Host operating system
Windows 10 Pro

### Guest operating system
Ubuntu 16.04

### Vagrantfile
Not convinced that it's relevant.

### Expected behavior
```
==> homestead-7: Configuring and enabling network interfaces...
==> homestead-7: Mounting shared folders...
    homestead-7: /vagrant => C:/Users/Anon/Documents/Web/homestead
[proceeds as normal]
```

### Actual behavior
```
==> homestead-7: Mounting shared folders...
    homestead-7: /vagrant => C:/Users/Anon/Documents/Web/homestead
Vagrant was unable to mount VirtualBox shared folders. This is usually
because the filesystem "vboxsf" is not available. This filesystem is
made available via the VirtualBox Guest Additions and kernel module.
Please verify that these guest additions are properly installed in the
guest. This is not a bug in Vagrant and is usually caused by a faulty
Vagrant box. For context, the command attempted was:

mount -t vboxsf -o uid=1000,gid=1000 vagrant /vagrant

The error output from the command was:

: No such file or directory
```

### Steps to reproduce
1. `vagrant add laravel/homestead`
2. Watch mounting fail.
3. The opposite of "Profit!"

### References
- #8359 
- https://forums.virtualbox.org/posting.php?mode=reply&f=6&t=82165

I just tore-out what little hair I have left arriving at the conclusion that upgrading VirtualBox from 5.1.14 to 5.1.16 broke Vagrant. I controlled all other variables and upgraded/downgraded VirtualBox twice to confirm that this is the culprit.

I realize that Homestead users are on their own, but given this issue, #8359 , and the duplicates of both , I'm not convinced that this is a problem with Homestead.

Furthermore, the exact same Homestead version and configuration works perfectly well with VirtualBox 5.1.14, so I see no reason to spend more time reproducing this with "vanilla Vagrant".

To be clear, the underlying change in VirtualBox, whatever it may be, doesn't break NFS mounts exclusively (per the issue referenced above). It breaks VirtualBox's native "Shared Folder" mounts, too, at least in whatever form Vagrant is leveraging them.

With VirtualBox 5.1.14, I have no issues. Provisioning flies right past the `Mounting shared folders` step.

But the moment I upgrade to VirtualBox 5.1.16, any form of mounting fails.

As a last resort, I tried installing the VirtualBox Guest Additions manually and rebuilding the kernel modules; the error doesn't change.

The fix is clear-cut: downgrade to VirtualBox 5.1.14 and `vagrant reload --provision`. It fixes the issue without fail.
